core/ulog/builder.go

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package ulog
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"go.uber.org/fx/core/config"
+
+	"github.com/uber-go/zap"
+)
+
+// Configuration for logging with uberfx
+type Configuration struct {
+	Level   string
+	File    *FileConfiguration
+	Stdout  bool
+	Verbose bool
+
+	prefixWithFileLine *bool `yaml:"prefix_with_fileline"`
+	TextFormatter      *bool // use TextFormatter (default json)
+}
+
+// FileConfiguration describes the properties needed to log to a file
+type FileConfiguration struct {
+	Enabled   bool
+	Directory string
+	FileName  string
+}
+
+// LogBuilder is the struct containing logger
+type LogBuilder struct {
+	log       zap.Logger
+	logConfig Configuration
+	level     zap.Level
+}
+
+var _development = strings.Contains(config.GetEnvironment(), "development")
+
+// NewBuilder creates an empty builder for building ulog.Log object
+func NewBuilder(keyvals ...interface{}) *LogBuilder {
```

Comment:
I think this can be called simply `Builder()`. It is not crucial for the end user if the builder is "new" or reused, and makes for more concise code: `ulog.Builder().Build()`